pypesto/optimize/optimizer.py

```diff
@@ -630,3 +636,162 @@ def minimize(
 
     def is_least_squares(self):
         return False
+
+
+class NLoptOptimizer(Optimizer):
+    """
+    Global/Local optimization using NLopt.
+    Package homepage: https://nlopt.readthedocs.io/en/latest/
+    """
+
+    def __init__(self, method=None, local_method=None, options: Dict = None,
+                 local_options: Dict = None):
+        """
+        Parameters
+        ----------
+        method:
+            Local or global Optimizer to use for minimization.
+
+        local_method:
+            Local method to use in combination with the global optimizer (
+            for the MLSL family of solvers) or to solve a subproblem (for the
+            AUGLAG family of solvers)
+
+        options:
+            Optimizer options. scipy option `maxiter` is automatically
+            transformed into `maxeval` and takes precedence.
+
+        local_options:
+            Optimizer options for the local method
+        """
+
+        super().__init__()
+
+        if options is None:
+            options = {}
+        elif 'maxiter' in options:
+            options['maxeval'] = options.pop('maxiter')
+        if local_options is None:
+            local_options = {}
+        self.options = options
+        self.local_options = local_options
+        if nlopt is None:
+            raise ImportError(
+                "This optimizer requires an installation of NLopt. You can "
+                "install NLopt via pip install nlopt.")
```

Comment:
Is checked twice, here and in `minimize()`.

Reply:
True, if we check in `__init__`, it should be available in `minimize`.